### PR TITLE
[[ SB Inclusions ]] Allow script library inclusions on mobile platforms

### PR DESCRIPTION
**Pull request by livecodeali**
Builds on https://github.com/livecode/livecode/pull/3999

### CODE REVIEW

File: ide-support/revsblibrary.livecodescript [hunk 1]
```diff
@@ -1719,3 +1719,80 @@ command revSBResolveIncludedExternal pPlatform, @xFile
...
+      if there is a stack tLibraryStack then
+         appendToStringList the effective filename of stack tLibraryStack, tAuxiliaryStackfiles
+         appendToStringList "revInternal__LoadLibrary" && tLibraryStack, tStartupScript
+      end if
+   end repeat
+   
+   -- If the datagrid is used, then include the library as an aux stack
+   if "data grid templates" is in the subStacks of stack pStack and there is a stack "revDataGridLibrary" then
+      appendToStringList the effective filename of stack "revDataGridLibrary", tAuxiliaryStackFiles
+   end if
```
Comment (peter-b) [inline ide-support/revsblibrary.livecodescript]: Is there any possibility that the new inclusions abstraction could be used to remove this special case in the future (not in this PR)?

**Comment by peter-b** [comment]
This all looks good subject to merging #3999 and adding a release note!

File: ide-support/revsblibrary.livecodescript (hunk 1 above)
Comment (livecodeali) [inline ide-support/revsblibrary.livecodescript]: I suppose the way to think of it is as a custom control with a support library, so it should probably be something stated in a 'manifest' for the data grid control. Ideally we would handle it in the same way as any other custom control package once we have a package manager.

**Comment by livecodeali** [comment]
@livecode-vulcan review by @peter-b ok 5684470

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 568447069b84c60ddac4ac2ef2cb84232d46fd13

**Comment by livecode-vulcan** [comment]
😎 test success 568447069b84c60ddac4ac2ef2cb84232d46fd13
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2497)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2507)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2490)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2508)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2514)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2518)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2530)

**Comment by runrevmark** [comment]
This is cool - but we should probably be careful with regards libURL as it has slightly different semantics than the in-built networking stuff on iOS and Android. I _think_ the current implementation with the libURL engine messages means that libUrl will override the libUrl implementation... We should probably double check this before putting it into a release...

**Comment by livecodeali** [comment]
@runrevmark Yes you're right - It only called the engine implementation if the getUrl (or equivalent) message is passed. What's the best way of dealing with that then? Make it not available on mobile? Or just warn if it is included?

**Comment by livecodeali** [comment]
@livecode-vulcan review not ok 5684470

**Comment by livecode-vulcan** [comment]
💔 review by @livecodeali not ok 568447069b84c60ddac4ac2ef2cb84232d46fd13

**Comment by livecodeali** [comment]
Requires https://github.com/livecode/livecode-ide/pull/1171

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 1c0d2a9

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 1c0d2a962ede8c3d70f6a4f06b7ae8ae9f2f3353

**Comment by livecode-vulcan** [comment]
😎 test success 1c0d2a962ede8c3d70f6a4f06b7ae8ae9f2f3353
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2516)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2526)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2511)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2527)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2533)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2537)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2549)

**Pull request merged**